processor/schemaprocessor/internal/translation/provider.go

```diff
@@ -0,0 +1,73 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package translation // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/schemaprocessor/internal/translation"
+
+import (
+	"bytes"
+	"context"
+	"embed"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+// Provider allows for collector extensions to be used to look up schemaURLs
+type Provider interface {
+	// Lookup whill check the underlying provider to see if content exists
+	// for the provided schemaURL, in the even that it doesn't an error is returned.
+	Lookup(ctx context.Context, schemaURL string) (content io.Reader, err error)
```

Comment:
Can we rename this to Retrieve (tldr , make it consistent with confmap provider https://github.com/open-telemetry/opentelemetry-collector/blob/83d93cd7cf86b00ef6783365c127ebe4b33f06af/confmap/provider.go#L74C2-L74C10 )

Reply:
Can we return a string or a struct with result embedded